crates/polars-sql/src/context.rs

```diff
@@ -600,44 +606,54 @@ impl SQLContext {
         lf = self.process_where(lf, &select_stmt.selection)?;
 
         // Column projections.
-        let projections: Vec<_> = select_stmt
+        let projections: Vec<Expr> = select_stmt
             .projection
             .iter()
             .map(|select_item| {
                 Ok(match select_item {
-                    SelectItem::UnnamedExpr(expr) => parse_sql_expr(expr, self, schema.as_deref())?,
+                    SelectItem::UnnamedExpr(expr) => {
+                        vec![parse_sql_expr(expr, self, schema.as_deref())?]
+                    },
                     SelectItem::ExprWithAlias { expr, alias } => {
                         let expr = parse_sql_expr(expr, self, schema.as_deref())?;
-                        expr.alias(&alias.value)
+                        vec![expr.alias(&alias.value)]
                     },
-                    SelectItem::QualifiedWildcard(oname, wildcard_options) => self
-                        .process_qualified_wildcard(
-                            oname,
+                    SelectItem::QualifiedWildcard(obj_name, wildcard_options) => {
+                        let expanded = self.process_qualified_wildcard(
+                            obj_name,
                             wildcard_options,
                             &mut contains_wildcard_exclude,
-                        )?,
+                            schema.as_deref(),
+                        )?;
+                        rewrite_projections(vec![expanded], &(schema.clone().unwrap()), &[])?
```

Comment:
Instead of making `rewrite_projections` public, we should ensure the projections get expanded in the operation that accepts this. Do you know which lazy_frame node this gets to?

Reply:
Hmm, I think the issue here is that I needed access to the expanded columns _before_ they were actually projected... If it's an issue then l'll see if I can rework things to create and deal with `vec[col]` instead of `cols` (which is where I think the need for this came from) 🤔

Reply:
Ah.. yes. We should not be relying om magic here too much. Especially since we need to lower to IR later. 🤞

Reply:
Halfway there; _mostly_ working without use of `rewrite_proejctions` but am now playing "whack-a-mole" with all the side effects of removing `cols` usage. Will get there in the end ;)

Reply:
Done 😅 The `sql` crate now handles all expansions without the use of `rewrite_projections`. While the mechanics of doing so got a bit more involved, having local access to expanded projections allowed for _significant_ simplification of wildcard exclusions (which then led to supporting some additional select-wildcard options)